ci/make_dev_version.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+# Copyright (c) 2020 Radio Astronomy Software Group
+# Licensed under the 2-clause BSD License
+
+# This function is make_dev_version.sh
+#
+# Usage:
+#   ./ci/make_dev_version.sh
+#
+# This function *must* be called from the main package folder where the
+# `setup.py` file lives. It echoes a modified version of the package which
+# includes the standard setuptools_scm "number of commits since last release"
+# counter, *plus* a numericized version of the git hash. This allows for uploads
+# to the PyPI test server that do not collide for branches that are both the
+# same number of commits since the last release, but have unique git hashes.
+#
+# This uses the "main" part of the version and the local git hash to generate a
+# version number that is acceptable to the testpypi server. According to PEP
+# 440, only numeric values are allowed in the ".dev" field, and testpypi does
+# not allow for "local" information (such as the git version). To get around
+# these constraints, we substitute the hex characters in the git hash for the
+# corresponding numeric values (e.g., a -> 10, b -> 11, etc.). We also prepend
+# numbers with a leading "0" (e.g., 1 -> 01) to avoid hash collisions.
+#
+# As an added benefit, this approach also handles tagged release versions (such
+# as "2.0.3" with no trailing "dev" information), and does not mangle the
+# version information.
```

Comment:
Would be really nice to have a few examples here, eg

```
# v1.0.2.dev23+g1af3e56 --> v1.0.2dev2301101503140506
# v1.0.2 --> v1.0.2
```

Reply:
Good suggestion! I'll add a few.